src/Behat/Behat/Context/Environment/InitializedContextEnvironment.php

```diff
@@ -35,8 +37,7 @@ final class InitializedContextEnvironment implements ContextEnvironment, Service
     private ?ContainerInterface $serviceContainer = null;
 
     /**
-     * @var array<class-string<Context>, Context>
-     * @psalm-var class-string-map<T as Context, T>
+     * @var array<class-string<T>, T>
```

Comment:
This is the equivalent of what we were achieving previously with the Psalm annotation: declaring that this is an array where the keys are class strings of Context classes and the values are instances of the Context class defined in the key

Reply:
This is not the same than before. The `class-string-map` was saying that each value of the array had to be an instance of the class used as key. The type you use here says that all keys must be class names extending the base type `T` and and values must be instances of that `T`, without any relation between keys and values.